springfox-core/src/main/java/springfox/documentation/service/ResolvedMethodParameter.java

```diff
@@ -69,18 +82,45 @@ public boolean hasParameterAnnotation(Class<? extends Annotation> annotation) {
   public int getParameterIndex() {
     return parameterIndex;
   }
+  
+  public boolean isReturnType() {
+    return returnType;
+  }
 
   public Optional<String> defaultName() {
     return Optional.fromNullable(defaultName);
   }
 
   public ResolvedMethodParameter replaceResolvedParameterType(ResolvedType parameterType) {
-    return new ResolvedMethodParameter(parameterIndex, defaultName, annotations, parameterType);
+    return new ResolvedMethodParameter(parameterIndex, returnType, defaultName, annotations, parameterType);
   }
 
   public ResolvedMethodParameter annotate(Annotation annotation) {
     List<Annotation> annotations = newArrayList(this.annotations);
     annotations.add(annotation);
-    return new ResolvedMethodParameter(parameterIndex, defaultName, annotations, parameterType);
+    return new ResolvedMethodParameter(parameterIndex, returnType, defaultName, annotations, parameterType);
+  }
+ 
+  @Override
+  public int hashCode() {
+    return Objects.hashCode(parameterIndex, returnType, defaultName, parameterType);
+  }
+  
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    ResolvedMethodParameter that = (ResolvedMethodParameter) o;
+    
+    return Objects.equal(parameterIndex, that.parameterIndex) &&
+        Objects.equal(returnType, that.returnType) &&   
+        Objects.equal(defaultName, that.defaultName) &&
+        Objects.equal(parameterType, that.parameterType);
```

Comment:
Same as above. I'd rather extract this to an `Equivalence` [like this](https://github.com/springfox/springfox/blob/master/springfox-spring-web/src/main/java/springfox/documentation/spring/web/OperationCachingEquivalence.java).